utilities/cli-runner/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/main.kt

```diff
@@ -40,5 +42,11 @@ private fun mainImpl(args: Array<String>, konancMain: (Array<String>) -> Unit) {
 
 fun main(args: Array<String>) = mainImpl(args, ::konancMain)
 
-fun daemonMain(args: Array<String>) = mainImpl(args, ::konancMainForGradle)
+private fun setupClangEnv() {
+    setEnv("LIBCLANG_DISABLE_CRASH_RECOVERY", "1")
+}
 
+fun daemonMain(args: Array<String>) {
+    usingNativeMemoryAllocator { setupClangEnv() }
```

Comment:
It would be great to avoid additional `usingNativeMemoryAllocator`: it allocates a huge block of memory here.

Can `setupClangEnv` be moved to `prepareTool`, just before `System.load`? This would also probably allow us to drop this environment from CLI scripts and other places eventually.
On the other hand, in this case we would disable any possibility to avoid running `setupClangEnv`, so if it has a problem and fails for someone we won't be able to provide a workaround. But I guess making it more like `if (has no env) setEnv()` should be safe enough.

Alternatively, to eliminate redundant `usingNativeMemoryAllocator` here we should consider wrapping the entire `main` and `daemonMain` to `usingNativeMemoryAllocator` (and remove it from other places).